pkg/controller/jobs/job/job_multikueue_adapter.go

```diff
@@ -170,10 +170,6 @@ func (b *multiKueueAdapter) DeleteRemoteObject(ctx context.Context, remoteClient
 	return client.IgnoreNotFound(remoteClient.Delete(ctx, &job, client.PropagationPolicy(metav1.DeletePropagationBackground)))
 }
 
-func (b *multiKueueAdapter) KeepAdmissionCheckPending() bool {
-	return !features.Enabled(features.MultiKueueBatchJobWithManagedBy)
```

Comment:
Let me double-check: can this be done without any change in the Job controller?

For Pods, you're changing the logic in `pod_controller.go` - so that promoting the MK AC from "Pending" to "Ready" will not prematurely activate the Pod.
So I imagined that, while touching the code here, you'd do sth analogous for Jobs (at least when the FG is disabled) - but I don't see it done.

(And, to my confusion, I saw one comment from you about moving the logic `(p *Pod) Run(...)` to somewhere else - matching what I imagined you'd do - however, that comment got later reworded, and the logic seems not moved in the end. Can you explain what happened? Or maybe I'm seeing old code due to a missing push?)